src/main/java/com/palantir/docker/compose/DockerService.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ */
+
+package com.palantir.docker.compose;
+
+import com.palantir.docker.compose.connection.waiting.SingleServiceHealthCheck;
+import org.immutables.value.Value;
+
+@Value.Immutable
+public interface DockerService {
```

Comment:
I'm not sure I would call the combination of a service name and healthcheck a `DockerService`? Is this only to make immutables work? There is a sneaky way to write custom methods for your builders.